web/index.ts

```diff
@@ -1,419 +1,430 @@
-                })),
-                H(Field, {
-                    label: `Image ${images.length + 1}`,
-                    input: H(Button, {
-                        label: `Add Image ${images.length + 1}`,
-                        onclick: () => {
-                            const nextImage = images.length === 1
-                                ? 'https://cdn.jsdelivr.net/gh/remojansen/logo.ts@master/ts.svg'
-                                : '';
-                            setLoadingState({ images: [...images, nextImage] })
-                        }
-                    }),
-                }),
+    setState({ ...newState, loading: true });
+  };
+  const {
+    fileType = "png",
+    fontSize = "75px",
+    theme = "light",
+    md = true,
+    text = "**Hello** World",
+    images = [imageLightOptions[0].value],
+    widths = [],
+    heights = [],
+    showToast = false,
+    messageToast = "",
+    loading = true,
+    selectedImageIndex = 0,
+    overrideUrl = null,
+  } = state;
+  const mdValue = md ? "1" : "0";
+  const imageOptions = theme === "light" ? imageLightOptions : imageDarkOptions;
+  const url = new URL(window.location.origin);
+  url.pathname = `${encodeURIComponent(text)}.${fileType}`;
+  url.searchParams.append("theme", theme);
+  url.searchParams.append("md", mdValue);
+  url.searchParams.append("fontSize", fontSize);
+  for (let image of images) {
+    url.searchParams.append("images", image);
+  }
+  for (let width of widths) {
+    url.searchParams.append("widths", width);
+  }
+  for (let height of heights) {
+    url.searchParams.append("heights", height);
+  }
+
+  return H(
+    "div",
+    { className: "split" },
+    H(
+      "div",
+      { className: "pull-left" },
+      H(
+        "div",
+        H(Field, {
+          label: "Theme",
+          input: H(Dropdown, {
+            options: themeOptions,
+            value: theme,
+            onchange: (val: Theme) => {
+              const options = val === "light" ? imageLightOptions : imageDarkOptions;
+              let clone = [...images];
+              clone[0] = options[selectedImageIndex].value;
+              setLoadingState({ theme: val, images: clone });
+            },
+          }),
+        }),
+        H(Field, {
+          label: "File Type",
+          input: H(Dropdown, {
+            options: fileTypeOptions,
+            value: fileType,
+            onchange: (val: FileType) => setLoadingState({ fileType: val }),
+          }),
+        }),
+        H(Field, {
+          label: "Font Size",
+          input: H(Dropdown, {
+            options: fontSizeOptions,
+            value: fontSize,
+            onchange: (val: string) => setLoadingState({ fontSize: val }),
+          }),
+        }),
+        H(Field, {
+          label: "Text Type",
+          input: H(Dropdown, {
+            options: markdownOptions,
+            value: mdValue,
+            onchange: (val: string) => setLoadingState({ md: val === "0" }),
```

Comment:
```suggestion
            onchange: (val: string) => setLoadingState({ md: val === "1" }),
```

Reply:
I very much appreciate you taking the time to check in on stream and help me out :)